File: kafka-helper-domain/src/main/java/com/hg/kafka/helper/domain/entity/DeadLetter.java
package com.hg.kafka.helper.domain.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;

@Document("dead_letter")
@Data
public class DeadLetter {
    @Id
    private String id;
    @Field("msg_id")
    private String messageId;
    @Field("msg")
    private String messageJson;
    @Field("reason")
    private String reason;
    @Field("topic")
    private String topic;
    @Field("status")
    private String status;
    @Field("partition")
    private Integer partition;
    @Field("offset")
    private Long offset;
    @Field("env")
    private String environment;
    @Field("created_at")
    private Date createdAt;
    @Field("updated_at")
    private Date updatedAt;
    @Field("trace_id")
    private String kafkaTraceId;
    @Field("group_id")
    private String groupId;

}

File: kafka-helper-domain/src/main/kotlin/com/hg/kafka/helper/domain/entity/DeadLetter.kt
package com.hg.kafka.helper.domain.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import java.util.*

@Document("dead_letter")
data class DeadLetter(
        @Id var id: String? = null,
        @Field("msg_id") var messageId: String,
        @Field("msg") var messageJson: String,
        @Field("reason") var reason: String,
        @Field("topic") var topic: String,
        @Field("status") var status: String,
        @Field("partition") var partition: Int,
        @Field("offset") var offset: Long,
        @Field("env") var environment: String,
        @Field("created_at") var createdAt: Date,
        @Field("updated_at") var updatedAt: Date,
        @Field("trace_id") var kafkaTraceId: String,
        @Field("group_id") var groupId: String
)
